prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/vendors-page/vendors-page.component.ts

```diff
@@ -103,7 +111,14 @@ export class VendorsPageComponent implements OnInit {
       .subscribe(({ vendors }: HealthAuthority) =>
         (vendors?.length)
           ? this.configService.vendors
-            .filter(v => vendors.some(({ vendorCode }) => v.code === vendorCode))
+            .reduce((vendorConfigs: VendorConfig[], v: VendorConfig) => {
```

Comment:
this block can be simplified by looping through `vendors` only, but will require changes to `VendorConfig` which in turn can have a regression impact. It may be easier to create a `HealthAuthorityVendorConfig`

Reply:
Yeah should be making another model and then when you initialize the filteredVendors subject you convert the vendorConfigs into HealthAuthorityVendors and then only use HealthAuthorityVendors in this component.
Same goes with HealthAuthorityCareTypes

Reply:
After fiddling with this, our `vendors` do not have a **vendor name**, therefore we need to make changes to the **VendorVM** and **Model** and add a migration, we'd also have to change the endpoints to take a list of **VendorVMs** rather than just **vendorCodes** in order to have the vendor name in the DB when we call the HA vendor update mehtod. Or we could keep the way that it is mapping the name to the rest of the object?

Reply:
Like we were discussing a private member variable of `private vendorIdCodeMap: { id: number, code: number }` and then using it remove would de-complicate this quite a bit.

Reply:
Fixed